server/aesdsocket.c

```diff
@@ -282,134 +280,320 @@ int listenForConnections(int sockfd, socket_data_t **newListElement)
+    char tempStr[sizeof(IOCSEEK_CMD_STR) / sizeof(IOCSEEK_CMD_STR[0])];
+    long ioctlRes;
+    int i;
+    bool tempBuffFlushed = true;
+    off_t retval;
+
+    // Append each byte received to the output file
+    while (recv(socket_data->connectedSock, &recvdByte, 1, 0) != 0)
     {
-        if(!mutexLocked)
+        if (!mutexLocked)
         {
 
             lockRet = pthread_mutex_lock(&mutex);
 
-            if(lockRet!=0)
+            if (lockRet != 0)
             {
                 perror("pthread_mutex_lock() error");
-                socket_data->threadCompleteFlag = true;
+                command_parser_state = NOT_CMD;
                 pthread_exit(socket_data);
             }
 
-            mutexLocked=true;
-
+            mutexLocked = true;
         }
 
-        while(write(outputFd, &recvdByte, 1)!=1)
-            ;
-
-        //Output contents of output file to
-        //the peer for every packet received
-        if(recvdByte=='\n')
+        switch (command_parser_state)
         {
-#ifndef USE_AESD_CHAR_DEVICE
-            //Set file pointer to 0
-            if(lseek(outputFd, 0, SEEK_SET)==-1)
+        case HEADER:
+            if (recvdByte != IOCSEEK_CMD_STR[commandStrInd])
             {
-                perror("lseek() error in returning socket input"
-                    "to peer");
-                    close(outputFd);
-                    //Include error element in socket_data_t?
-                    socket_data->threadCompleteFlag = true;
-                    pthread_exit(socket_data);
+                syslog(LOG_INFO, "Last character read: %c", recvdByte);
+                syslog(LOG_INFO, "Last character index: %d", commandStrInd);
+                command_parser_state = NOT_CMD;
+                tempStr[commandStrInd] = recvdByte;
+                tempStr[commandStrInd + 1] = '\0';
+                tempBuffFlushed = false;
             }
-#endif
-            
-            readRet=1;
-            //Read until EOF
-            while(readRet!=0)
+            else
+            {
+                tempStr[commandStrInd] = recvdByte;
+                commandStrInd++;
+                if (commandStrInd == (sizeof(IOCSEEK_CMD_STR) / sizeof(IOCSEEK_CMD_STR[0]) - 1))
+                {
+                    command_parser_state = ARG_X;
+                    commandStrInd = 0;
+                }
+            }
+            break;
+
+        case ARG_X:
+            if (recvdByte != ',')
+            {
+                argX[commandStrInd] = recvdByte;
+                commandStrInd++;
+            }
+            else
+            {
+                argX[commandStrInd] = '\0';
+                syslog(LOG_INFO, "Arg X: %s", argX);
+                command_parser_state = ARG_Y;
+                commandStrInd = 0;
+            }
+            break;
+
+        case ARG_Y:
+            if (recvdByte != '\n')
             {
-                do
+                argY[commandStrInd] = recvdByte;
+                commandStrInd++;
```

Comment:
Similar comments here, I think you need handling for the case where `commandStrInd` would overwrite array bounds on `argY`